modules/standard/CommDiagnostics.chpl

```diff
@@ -372,6 +372,81 @@ module CommDiagnostics
   }
 
 
+  /*
+    Print the current communication counts in a markdown table using a
+    row per locale and a column per operation.  By default, operations
+    for which all locales have a count of zero are not displayed in
+    the table, though an argument can be used to reverse that
+    behavior.
+
+    :arg printEmptyColumns: Indicates whether empty columns should be printed (defaults to ``false``)
+    :type printEmptyColumns: `bool`
+  */
+  proc printCommDiagnosticsTable(printEmptyColumns=false) {
+    use Reflection;
+    param unstable = "unstable";
+
+    // grab all comm diagnostics
+    var CommDiags = getCommDiagnostics();
+
+    // cache number of fields and store vector of whether field is active
+    param nFields = numFields(chpl_commDiagnostics);
+
+    // How wide should the column be for this field?  A negative value
+    // indicates an unstable field.  0 indicates that the field should
+    // be skipped in the table.
+    var fieldWidth: [0..<nFields] int;
+
+    // print column headers while determining which fields are active
+    writef("| %6s ", "locale");
+    for param fieldID in 0..<nFields {
+      param name = getFieldName(chpl_commDiagnostics, fieldID);
+      // We should be able to write this as follows:
+      //
+      //      const maxval = max reduce [locID in LocaleSpace] getField(CommDiags[locID], fieldID);
+      //
+      // except that it doesn't work due to #16042.  So I'm using this
+      // annoying workaround instead:
+
+      var maxval = 0;
+      for locID in LocaleSpace do
+        maxval = max(maxval, getField(CommDiags[locID], fieldID).safeCast(int));
+
+      if printEmptyColumns || maxval != 0 {
+        const width = if commDiagsPrintUnstable && name == "amo"
```

Comment:
(related to comment by @ronawho)

Backwards?  Think we want to print 'unstable' for AMOs iff `commDiagsPrintUnstable==false` (the default), and the actual value otherwise.  This would conform to `writeThis()`, except that the latter doesn't print anything at all for AMOs when `commDiagsPrintUnstable==false`.